cardshifter-core/src/main/java/com/cardshifter/core/groovy/EffectDelegate.groovy

```diff
@@ -257,10 +270,9 @@ class EffectDelegate {
                     })
                 }
             }
-            assert closure : "$description: Unknown target $who"
+            assert closure : "Unknown target $who"
```

Comment:
Providing additional information would be good. I actually managed to fire this assertion when doing this patch and was at first very confused as to what was wrong.

> `assert closure : "Effect must have a valid target, not '$who'. Description list: $descriptionList"

What about something like this?

There are a lot of other assertions in this file that does not provide this kind of context. When I run in to any of them I just look at the stack trace to find where the error lies. I understand this is not feasible for a modder though.